core/src/main/scala/org/apache/spark/ui/jobs/AllJobsPage.scala

```diff
@@ -427,17 +430,24 @@ private[ui] class JobDataSource(
     val formattedDuration = duration.map(d => UIUtils.formatDuration(d)).getOrElse("Unknown")
     val submissionTime = jobData.submissionTime
     val formattedSubmissionTime = submissionTime.map(UIUtils.formatDate).getOrElse("Unknown")
-    val lastStageAttempt = store.lastStageAttempt(jobData.stageIds.max)
-    val lastStageDescription = lastStageAttempt.description.getOrElse("")
-
+    var lastStageDescription = ""
```

Comment:
Instead of catching the exception the logic should be modified to be prepared for a missing stageAttempt.